app/src/standard/java/com/twilio/voice/quickstart/DialpadActivity.java

```diff
@@ -0,0 +1,106 @@
+package com.twilio.voice.quickstart;
+
+import android.os.Bundle;
+import androidx.appcompat.app.AppCompatActivity;
+import android.media.AudioManager;
+import android.media.ToneGenerator;
+import android.widget.Button;
+import android.widget.EditText;
+import android.util.Log;
+import com.twilio.voice.Call;
+
+public class DialpadActivity extends AppCompatActivity {
+    private ToneGenerator toneGenerator;
+    private EditText dialpadEditText;
+    private final int[] digitIds = {R.id.button0, R.id.button1, R.id.button2, R.id.button3, R.id.button4, R.id.button5, R.id.button6, R.id.button7, R.id.button8, R.id.button9, R.id.buttonStar, R.id.buttonHash};
+    private final String[] digits = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "*", "#"};
+    private final int DTMF_TONE_VOLUME = 80;
+    private final int DTMF_TONE_PLAYBACK_DURATION_MS = 150;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.dialpad);
+
+        dialpadEditText = findViewById(R.id.dialpadEditText);
+        Button buttonDone = findViewById(R.id.buttonDone);
+
+        toneGenerator = new ToneGenerator(AudioManager.STREAM_VOICE_CALL, DTMF_TONE_VOLUME);
+
+        for (int i = 0; i < digitIds.length; i++) {
```

Comment:
This can be avoided by embedding a callback in the xml using ` android:onClick="<method name>" `.
https://developer.android.com/training/constraint-layout/motionlayout/ref/onclick